src/main/java/com/sap/cloudfoundry/client/facade/oauth2/TokenFactory.java

```diff
@@ -0,0 +1,83 @@
+package com.sap.cloudfoundry.client.facade.oauth2;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Base64.Decoder;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.web.server.ResponseStatusException;
+
+import com.sap.cloudfoundry.client.facade.util.JsonUtil;
+
+public class TokenFactory {
+
+    // Scopes:
+    public static final String SCOPE_CC_READ = "cloud_controller.read";
+    public static final String SCOPE_CC_WRITE = "cloud_controller.write";
+    public static final String SCOPE_CC_ADMIN = "cloud_controller.admin";
+    public static final String SCOPE_SCIM_USERIDS = "scim.userids";
+    public static final String SCOPE_PASSWORD_WRITE = "password.write";
+    public static final String SCOPE_OPENID = "openid";
+
+    // Token Body elements:
+    public static final String SCOPE = "scope";
+    public static final String EXP = "exp";
+    public static final String IAT = "iat";
+    public static final String USER_NAME = "user_name";
+    public static final String USER_ID = "user_id";
+    public static final String CLIENT_ID = "client_id";
+
+    public OAuth2AccessTokenWithAdditionalInfo createToken(String tokenString) {
+        Map<String, Object> tokenInfo = parseToken(tokenString);
+        return createToken(tokenString, tokenInfo);
+    }
+
+    @SuppressWarnings("unchecked")
+    public OAuth2AccessTokenWithAdditionalInfo createToken(String tokenString, Map<String, Object> tokenInfo) {
+        List<String> scope = (List<String>) tokenInfo.get(SCOPE);
+        Number expiresAt = (Number) tokenInfo.get(EXP);
+        Number instantiatedAt = (Number) tokenInfo.get(IAT);
+        if (scope == null || expiresAt == null || instantiatedAt == null) {
+            return null;
+        }
+        return new OAuth2AccessTokenWithAdditionalInfo(createOAuth2AccessToken(tokenString, scope, expiresAt, instantiatedAt), tokenInfo);
+    }
+
+    private OAuth2AccessToken createOAuth2AccessToken(String tokenString, List<String> scope, Number expiresAt, Number instantiatedAt) {
+        try {
+            return new OAuth2AccessToken(OAuth2AccessToken.TokenType.BEARER,
+                                         tokenString,
+                                         new Date(instantiatedAt.longValue() * 1000).toInstant(),
```

Comment:
Why not use `Instant.ofEpochSeconds()` if the `instancedAt` and `expiresAt` variables are in seconds?